doc/globbing.rdoc

```diff
@@ -0,0 +1,356 @@
+    pathname.glob(pattern).take(3)
+    # => [#<Pathname:array.rb>, #<Pathname:ast.rb>, #<Pathname:autogen.sh>]
+
+- <tt>'\'</tt>: escapes the next metacharacter:
+
+  Note that on Windows, the backslash character may not be used in a string pattern:
+  <tt>Dir['c:\foo*']</tt> will not work, use <tt>Dir['c:/foo*']</tt> instead.
+
+See also flag File::FNM_EXTGLOB below.
+
+=== Flags
+
+\Integer _flags_ may be used to restrict, expand, or otherwise modify
+the behavior of the filename globbing methods:
+
+- Dir.glob and Pathname.glob accept optional keyword argument +flags+
+  with an integer value:
+
+    Dir.glob('*', flags: File::FNM_DOTMATCH).take(5)
+    # => [".", ".appveyor.yml", ".cirrus.yml", ".dir-locals.el", ".document"]
+    Pathname.glob('*', flags: File::FNM_DOTMATCH).take(3)
+    # => [#<Pathname:.>, #<Pathname:.appveyor.yml>, #<Pathname:.cirrus.yml>]
+
+- Pathname#glob accepts an optional second argument as a flags value:
+
+    pathname.glob('*', File::FNM_DOTMATCH).take(3)
+    # => [#<Pathname:.>, #<Pathname:.appveyor.yml>, #<Pathname:.cirrus.yml>]
+
+The flags value may be the bitwise OR of any combination of individual
+flag constants;  example:
+
+  flags = File::FNM_CASEFOLD | File::FNM_DOTMATCH
+
+==== Flag File::FNM_CASEFOLD
+
+The FNM_CASEFOLD flag specifies that the matching is to be case-insensitive:
+
+  Dir.glob('c*').take(5)
+  # => ["CONTRIBUTING.md", "COPYING", "COPYING.ja", "ccan", "class.c"]
+  Dir.glob('c*', flags: File::FNM_CASEFOLD).take(5)
+  # => ["CONTRIBUTING.md", "COPYING", "COPYING.ja", "ccan", "class.c"]
+
+==== Flag File::FNM_DOTMATCH
+
+The FNM_DOTMATCH flag specifies that entry names beginning with <tt>'.'</tt>
+should be considered for matching:
+
+  Dir.glob('*').take(5)
+  # => ["BSDL", "CONTRIBUTING.md", "COPYING", "COPYING.ja", "GPL"]
+  Dir.glob('*', flags: File::FNM_DOTMATCH).take(5)
+  # => [".", ".appveyor.yml", ".cirrus.yml", ".dir-locals.el", ".document"]
+
+==== Flag File::FNM_EXTGLOB
+
+Flag File::FNM_EXTGLOB flag enables the pattern extension
+<tt>'{_a_,_b_}'</tt>, which matches pattern _a_ and pattern _b_;
+  behaves like
+  a {regexp union}[https://docs.ruby-lang.org/en/master/Regexp.html#method-c-union]
+  (e.g., <tt>'(?:_a_|_b_)'</tt>):
+
+    pattern = '{LEGAL,BSDL}'
+    Dir.glob(pattern)      # => ["LEGAL", "BSDL"]
+    Pathname.glob(pattern) # => [#<Pathname:LEGAL>, #<Pathname:BSDL>]
+    pathname.glob(pattern) # => [#<Pathname:LEGAL>, #<Pathname:BSDL>]
+
+==== Flag File::FNM_NOESCAPE
+
+Flag File::FNM_NOESCAPE specifies that escaping with the backslash character <tt>'\'</tt>
+is disabled; the character is not an escape character.
+
+==== Flag File::FNM_PATHNAME
+
+Metacharacters <tt>'*'</tt> and <tt>'?'</tt> do not match directory separators.
+
+==== Flag File::FNM_SHORTNAME
+
+Patterns may match short names if they exist; Windows only.
+
+==== Flag File::FNM_SYSCASE
+
+Case sensitivity (sensitive or insensitive) is the same as the system.
```

Comment:
We should note that it's also only useful for `File.fnmatch` and not any other methods.